File: product-brain/src/product_brain/bot/commands.py
from __future__ import annotations

import re
from dataclasses import dataclass
from typing import Optional


_BRAIN_RE = re.compile(r"(?:^|\n)\s*/brain(?:\s+(\S+))?(?:\s+(.*))?")

VALID_COMMANDS = {
    "groom", "estimate", "edges", "related",
    "draft-tickets", "refresh", "explain", "on", "off",
}


@dataclass
class BrainCommand:
    command: str
    args: str = ""


def parse_brain_command(comment_body: str) -> Optional[BrainCommand]:
    if not comment_body:
        return None
    m = _BRAIN_RE.search(comment_body)
    if not m:
        return None
    cmd = (m.group(1) or "groom").lower().strip()
    args = (m.group(2) or "").strip()
    if cmd not in VALID_COMMANDS:
        return None
    return BrainCommand(command=cmd, args=args)

File: product-brain-ts/src/bot/commands.ts

const BRAIN_RE = /(?:^|\n)\s*\/brain(?:\s+(\S+))?(?:\s+([^\n]*))?/;

export const VALID_COMMANDS = new Set([
  "groom",
  "estimate",
  "edges",
  "related",
  "draft-tickets",
  "refresh",
  "explain",
  "on",
  "off",
]);

export interface BrainCommand {
  command: string;
  args: string;
}

export function parseBrainCommand(commentBody: string): BrainCommand | null {
  if (!commentBody) return null;
  const m = BRAIN_RE.exec(commentBody);
  if (!m) return null;
  const cmd = (m[1] ?? "groom").toLowerCase().trim();
  const args = (m[2] ?? "").trim();
  if (!VALID_COMMANDS.has(cmd)) return null;
  return { command: cmd, args };
}
